native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe.Tests/deserializer/avro/AvroDeserializerTest.cs

```diff
@@ -0,0 +1,538 @@
+        }
+        
+        [Test]
+        public void TestDeserialize_AllTypes_Succeed()
+        {
+            const string avroRecordMapName = "meta";
+            const string keyName = "testKey";
+            
+            var mixedSchema = (RecordSchema) AvroSchemas.GetValueOrDefault("user3.avsc", null);
+            Assert.NotNull(mixedSchema);
+            var mixedRecord = new GenericRecord(mixedSchema);
+            
+            var dictionary = new Dictionary<string, object> { { keyName, 1L } };
+            var intList = new[] { 1 };
+
+            var enumSchema = mixedSchema["integerEnum"].Schema;
+            var enumSymbol = new GenericEnum((EnumSchema)enumSchema, "ONE");
+            
+            
+            mixedRecord.Add(avroRecordMapName, dictionary);
+            mixedRecord.Add("name", "Joe");
+            mixedRecord.Add("favorite_number", 1);
+            mixedRecord.Add(avroRecordMapName, dictionary);
+            mixedRecord.Add("listOfColours", intList);
+            mixedRecord.Add("integerEnum", enumSymbol);
+            
+            
+            var serializedData = AvroSerializer.Serialize(mixedRecord);
+            
+            var schemaObject = new GlueSchemaRegistrySchema("testAvroSchema", 
+                mixedSchema.ToString(), 
+                GlueSchemaRegistryConstants.DataFormat.AVRO.ToString());
+            
+            var avroDeserializer = new AvroDeserializer(_avroGenericConfiguration);
+            
+            var deserialize = avroDeserializer.Deserialize(serializedData,
+                schemaObject);
+            Assert.True(deserialize is GenericRecord);
+            var record = (GenericRecord) deserialize;
+            var deserializedMap = (Dictionary<string,object>)record.GetValue(record.Schema[avroRecordMapName].Pos);
+            Assert.AreEqual(dictionary.Keys.GetEnumerator().Current, 
+                deserializedMap.Keys.GetEnumerator().Current);
+            Assert.AreEqual(dictionary[keyName], deserializedMap[keyName]);
+            Assert.AreEqual("Joe", record.GetValue(record.Schema["name"].Pos));
+            Assert.AreEqual(1, record.GetValue(record.Schema["favorite_number"].Pos));
+            Assert.AreEqual(intList, record.GetValue(record.Schema["listOfColours"].Pos));
+            Assert.AreEqual(enumSymbol, record.GetValue(record.Schema["integerEnum"].Pos));
+        }
+        
+        /*
+         * Test Invalid record types 
+         */
+        
+        [Test]
+        public void TestDeserialize_UnknownRecordType_ThrowsException()
+        {
+            var serializedResult = AvroSerializer.Serialize(_userDefinedGenericRecord);
+            var avroDeserializer = new AvroDeserializer(_avroUnknownConfiguration);
+
+            var exception =Assert.Throws(typeof(AwsSchemaRegistryException), 
+                ()=>avroDeserializer.Deserialize(serializedResult,GenericRecordSchema));
```

Comment:
Minor: Formatting. Can we have a Linter setup for the package?

Reply:
Updated, I have added a linter and it will give us warning in build time for these kinds of issue. I will configure the linter and resolve warnings in a separate PR.